internal/pkg/db/db.go

```diff
@@ -104,31 +109,31 @@ func NewBackend(opts ...Option) (*Backend, error) {
 	return backend, nil
 }
 
-// AddDocument adds the protobom Document to the database and annotates with its source data, hash, and format.
-func (backend *Backend) AddDocument(sbomData []byte) (*sbom.Document, error) {
+// AddDocument adds the protobom document to the database and annotates with given annotations.
+func (backend *Backend) AddDocument(sbomData []byte, annotations ...Option) (*sbom.Document, error) {
+	// defer clearing annotations for next run
```

Comment:
nits:
- casing
- If the code has `defer` in it, the comment shouldn't need to restate that
- the wording could be a little more focused on "why is it doing this" rather than "what is it doing"
  - e.g. `// Clear annotations after adding document.` 

```suggestion
	// Defer clearing annotations for next run.
```